DXC preview fails to build

Preview failed to build: https://github.com/compiler-explorer/infra/actions/runs/6963684112/job/18949687274

```
  [417/1557] Building CXX object lib/IR/CMakeFiles/LLVMCore.dir/Core.cpp.o
  /root/dxc/lib/IR/Core.cpp: In function 'void LLVMContextSetDiagnosticHandler(LLVMContextRef, LLVMDiagnosticHandler, void*)':
  /root/dxc/lib/IR/Core.cpp:93:80: warning: cast between incompatible function types from 'LLVMDiagnosticHandler' {aka 'void (*)(LLVMOpaqueDiagnosticInfo*, void*)'} to 'llvm::LLVMContext::DiagnosticHandlerTy' {aka 'void (*)(const llvm::DiagnosticInfo&, void*)'} [-Wcast-function-type]```

_Originally posted by @mattgodbolt in https://github.com/compiler-explorer/infra/issues/1148#issuecomment-1823621724_

--- a/misc/build-dxc.sh
+++ b/misc/build-dxc.sh
@@ -16,6 +16,7 @@ URL=https://github.com/microsoft/DirectXShaderCompiler
 FULLNAME=dxc-${VERSION}.tar.xz
 OUTPUT=${ROOT}/${FULLNAME}
 S3OUTPUT=
+EXTRA_CMAKE_ARGS=
 if [[ $2 =~ ^s3:// ]]; then
     S3OUTPUT=$2
 else
@@ -38,6 +39,10 @@ if [[ "${REVISION}" == "${LAST_REVISION}" ]]; then
     exit
 fi
 
+if [[ "${VERSION}" == "1.8.2306-preview" ]]; then
+  EXTRA_CMAKE_ARGS="-DCMAKE_CXX_FLAGS=-fpermissive"
+fi
+
 BUILD_DIR=$(pwd)/dxc/build
 DIR=$(pwd)/dxc
 export PATH=${PATH}:/cmake/bin
@@ -45,7 +50,7 @@ git clone --recurse-submodules --depth 1 -b ${BRANCH} ${URL} ${DIR}
 
 cd ${DIR}
 mkdir -p ${BUILD_DIR}
-cmake -S . -B ${BUILD_DIR} -G Ninja -DCMAKE_BUILD_TYPE=Release -C ./cmake/caches/PredefinedParams.cmake
+cmake -S . -B ${BUILD_DIR} -G Ninja -DCMAKE_BUILD_TYPE=Release -C ./cmake/caches/PredefinedParams.cmake $EXTRA_CMAKE_ARGS
 cmake --build ${BUILD_DIR}
 
 export XZ_DEFAULTS="-T 0"

